custom_components/frigate/media_source.py

```diff
@@ -869,7 +869,7 @@ def _build_event_response(
                     title=f"{dt.datetime.fromtimestamp(event['start_time'], DEFAULT_TIME_ZONE).strftime(DATE_STR_FORMAT)} [{duration}s, {event['label'].capitalize()} {int(event['top_score']*100)}%]",
                     can_play=identifier.media_type == MEDIA_TYPE_VIDEO,
                     can_expand=False,
-                    thumbnail=f"data:image/jpeg;base64,{event['thumbnail']}",
+                    thumbnail=f"/api/frigate/{identifier.frigate_instance_id}/notifications/{event['id']}/thumbnail.jpg",
```

Comment:
This only works for users who have turned on the unauthenticated notifications proxy. Also, I can't remember if we ever implemented the time based restriction on that proxy.

Reply:
Oh I see. Is that the same for all of Frigate's http api?

Reply:
The option (`CONF_NOTIFICATION_PROXY_ENABLE`) is only to allow/disallow that _particular_ call, since it is (IIRC) the only unauthenticated one. Today that option just turns on whether requests to this endpoint are outright rejected immediately.

One possible path: we could change that option to toggling between allowing both authenticated and unauthenticated -- or just authenticated (since there's no other real reason you'd want to disable the proxy). Then we could make sure the users of this API always use the authenticated flavor, and then it would probably work whilst honoring the intent of that option ...

FTR There would be two end users of this API we need to consider:
1. The built-in HA media browser.
1. [The Frigate card](https://github.com/dermotduffy/frigate-hass-card)

Will the former automatically sign the URLs to result in the request that arrives back being authenticated? I don't think the latter will "just work" either -- each thumbnail URL would probably have to result in a separate call to HA to sign the URL in order to arrive authenticated (which might be a significant performance hit, since it would be one call per thumbnail, but would have to play around to see if that's the case...).

Reply:
> Also, I can't remember if we ever implemented the time based restriction on that proxy.

It's still stuck in a PR https://github.com/blakeblackshear/frigate-hass-integration/pull/169 , but still a good idea I think we should pursue (and so would be reluctant to need to rely on the unauthenticated endpoint for thumbnails).

Reply:
I think the frontend gets a signed URL: https://github.com/home-assistant/frontend/blob/c27e3325d98aff229277ce41698fd73d863d84e1/src/components/media-player/ha-media-player-browse.ts#L655

Reply:
Anyway, this is just an idea...can keep this on the backburner. It doesn't seem like there's much immediate benefit.

Reply:
I would love to avoid the base64. This is the only reason they are stored that way in Frigate's database, which adds bloat and overhead. I would rather just create a thumbnail file on disk.